CONTRIBUTING.md

```diff
@@ -32,6 +32,7 @@ expect throughout each step of the process.
       * [Commit message guidelines](#commit-message-guidelines)
     * [Step 5: Rebase](#step-5-rebase)
     * [Step 6: Test](#step-6-test)
+    * [Step 6.5: Test Coverage](#step-65-test-coverage)
```

Comment:
This section has great info. I'm not sure if CONTRIBUTING.md is the place for it. Can we move it to it's own document in `doc/guides` and link to it from CONTRIBUTING.md? CONTRIBUTING.md is the file people get a link asking them to read when they are in the process of opening a pull request.

Reply:
Yeah, wasn't quite sure where to put it ... I like the idea of a adding it to `writing-tests.md` or creating its own guide depending on what others are feeling.

Reply:
+1 to putting it in its own guide.